Modelica/Fluid/Examples/DrumBoiler.mo

```diff
@@ -70,9 +70,9 @@ package DrumBoiler
     Modelica.Blocks.Interfaces.RealOutput V_l(unit="m3") "Liquid volume inside drum"
       annotation (Placement(transformation(extent={{100,74},{112,86}})));
   public
-    Modelica.Blocks.Math.Gain MW2W(k=1e6)
+    Modelica.Blocks.Math.Gain MW2W(final k=1e6)
```

Comment:
Similar to `Pa2bar` but this should be combined with specifying the `unit = "MW"` for the table source:
```
  Modelica.Blocks.Sources.TimeTable q_F_Tab(table = [0, 0; 3600, 400; 7210, 400], y(unit = "MW")) if not use_inputs annotation(Placement(transformation(extent = {{-90, -80}, {-70, -60}})));
```
```suggestion
    Modelica.Blocks.Math.Gain MW2W(final k(unit = "W/MW", displayUnit = "1")  = 1e6)
```

Reply:
I might possibly accept the change of `unit`, but changing the `displayUnit` to "1" is "too cute" and a very bad idea.

The reason is that the display unit is used to input (and show) values in the display unit (after a conversion); that is the entire reason for having display units. In normal cases that is useful so that for `Real x(unit="W", displayUnit="MW")=1e6;` one would input/see "1 MW" (in various ways).

With `displayUnit="1"` it means that one should input a value in the display unit "1" and then have that converted to "W/MW", or more specifically input the value 1 in the display unit "1". It is cute that that the factor is exactly 1 for a conversion that in some sense doesn't change anything, but unless tools treat such cases specially in the GUI it will seem as if we are converting from "MW" to "W" using a conversion factor 1 (in unit "1").

Seeing a conversion factor 1 for something that clearly should convert between "MW" and "W" is just confusing, and not at all helpful!

And, if we are going to have special rules for such conversions it seems easier to say something "a variable used as multiplicative factor converting between different prefix-variants of the same unit should only do that using the correct factor, and not have any additional factor", or alternatively one could add a helper function for that (such as `conversion(1e-6, "MW", "W")`), instead of having a special rule for not converting and displaying in this specific case. That would fix the factor to the right value, no need for display unit or even unit.

The conversion-function would in one sense just return 1e-6 (with unit "W/MW"), but also check that it is the right factor, and it would be clearer that it is just converting - and not some weird correlation (like we lose 20 W in the cables per MW of transmitted power.)